File: src/trackers/es/puntotorrent_ch.ts
import {WebChecker} from "../../checker/webChecker";
import {CheckerResult} from "../../checker/checkerResult";

export default class Puntotorrent_ch extends WebChecker {
    public get url() { return "https://xbt.puntotorrent.ch"; }
    public get name() { return "puntotorrent.ch"; }
    public get language() { return "es"; }
    public get categories(): Array<String> { return ["movies"]; }
    public get description() { return ""; }
    public get registryUrl() { return "https://xbt.puntotorrent.ch/index.php?page=signup"; }

    async isOpen(): Promise<CheckerResult> {
        let $ = await WebChecker.getQueryUrl(this.registryUrl);
        let exists = $(":contains('Lo sentimos pero los registros')").length > 0;

        return {
            online: true,
            regClosed: !exists,
            regOpened: false
        };
    }
}

File: checker/trackers/puntotorrentch.go
package trackers

import (
	"github.com/tokkenno/open-torrent-tracker-bot/checker/check"
	"github.com/tokkenno/open-torrent-tracker-bot/models"
	"strings"
)

type PuntotorrentCh struct {
}

func (tracker *PuntotorrentCh) GetId() string {
	return "puntotorrent_ch"
}

func (tracker *PuntotorrentCh) GetName() string {
	return "PuntoTorrent"
}

func (tracker *PuntotorrentCh) GetDescription() string {
	return ""
}

func (tracker *PuntotorrentCh) GetLanguages() []string {
	return []string{"es"}
}

func (tracker *PuntotorrentCh) GetCategories() []string {
	return []string{"movies"}
}

func (tracker *PuntotorrentCh) GetUrl() string {
	return "https://xbt.puntotorrent.ch"
}

func (tracker *PuntotorrentCh) GetRegistryUrl() string {
	return "https://xbt.puntotorrent.ch/index.php?page=signup"
}

func (tracker *PuntotorrentCh) Check() *check.Result {
	html, err := check.GetWebCode(tracker.GetRegistryUrl())

	if err != nil {
		return &check.Result{
			IsOnline: false,
		}
	} else {
		var status models.RegistrationStatus

		if strings.Contains(html, "los registros están cerrados") {
			status = models.RegistrationClose
		} else {
			status = models.RegistrationMaybe
		}

		return &check.Result{
			IsOnline: true,
			Status:   status,
		}
	}
}
